### ISSUE TITLE
Level Emitter with crafting card doesnt work

### ISSUE THREAD

**Issue by DavidieHD** [AppliedEnergistics/Applied-Energistics-2#8251]
### Describe the bug

im playing ATM 10 and was trying to automate something with the level emitter and an crafting card in it. both the modis (emite while crafting and emite to craft). when an item is crafting and the emitter is in emite while crafting mode the redstone doesnt change. i tried it on my server when i reboot the system the emiter shortly activates and goes off after that, on an test single player world when i reboot the system the emitter stays on but when i cancel the autocrafting and start it again the emitter doenst do anything till i reboot the system. the same happens white the modi emite to craft.

### How to reproduce the bug

set an Level Emitter with an crafting card and the modi emite while crafting put an item in it and start auto crafting it.
or  with the modi emite to craft 

### Expected behavior

Level emitter should turn on when autocrafting this item

### Which minecraft version are you using?

1.21

### On which mod loaders does it happen?

NeoForge

### Crash log

https://pastebin.com/LEFjRrZh

**Comment by PiggyWhiskey** [comment]
I have the same issue in the same pack. It looks like it's a bad interaction between AdvancedAE and AE2.
On a simple test setup, I can get the level emitter to function. But once I use the Quantum Computer Core (256M + 12 Co-Processors) it ceases to function.

I think it's due to the "infinite" aspect of the Quantum Computer Core, where it can be practically unlimited Crafting Storage.

Davidie would have to confirm if he's using the Quantum core

**Comment by PiggyWhiskey** [comment]
on Advanced AE's Github:
https://github.com/pedroksl/AdvancedAE/issues/5

**Comment by DavidieHD** [comment]
> I have the same issue in the same pack. It looks like it's a bad interaction between AdvancedAE and AE2. On a simple test setup, I can get the level emitter to function. But once I use the Quantum Computer Core (256M + 12 Co-Processors) it ceases to function.
> 
> I think it's due to the "infinite" aspect of the Quantum Computer Core, where it can be practically unlimited Crafting Storage.
> 
> Davidie would have to confirm if he's using the Quantum core

yes i am indeed using the quantum computer core, i think i have testet with the normal cpu to but that didnt work either.

**Comment by DenysChuhlib** [comment]
I also play ATM10 and when using a quantum computer, only after updating the network does the emitter turn on (operation mode during crafting) for a couple of seconds and then turn off, but when using a regular processor (from AE2) it works normally.

**Comment by Mithi83** [comment]
I've confirmed that a simple setup (self-compiled AE2, advancedAE and some other mods from my dev environment) showed the bug in a simple network containing just one Pattern Provider with one Molecular Assembler, a pattern for Cable Anchors, a ME Level Emitter with Crafting Card set to Cable Anchors and infinite resources. I then switch between using a 256k Crafting Storage with 7 Co-Processors (setup A) or using a Quantum Computer Core (setup B) as my Crafting CPU.

Using setup A I can see that the ME Level Emitter triggers on and off frequently, whereas in setup B no trigger can be seen.

I then used a debugger to set a breakpoint in StorageLevelEmitterPart L122 (`craftingWatcherNode.onRequestChange`).
In setup A I could observe the breakpoint being triggered twice every time a craft is executed, e.g a craft containing 3 executions of the pattern results in 6 breakpoint triggers. My guess is that starting the pattern crafting in the Molecular Assembler triggers the first breakpoint and finishing the craft in the Molecular Assembler triggers the second breakpoint. In setup B the breakpoint is only triggered once after all crafts were completed. This discrepancy is likely a more low-level description of the bug.

The next step would be to find out how `ICraftingWatcherNode` works in detail and in particular what triggers it in the background. My intuition is that somehow the trigger is not sent at all or that the trigger is not sent because the on and off happens in the same tick and is therefore not recognized as a trigger event because the state before and after is the same.

**Comment by Mithi83** [comment]
I ended up in `CraftingService.onServiceEndTick`. I set a breakpoint in L160, inside the `if (latestChange != lastProcessedCraftingLogicChangeTick)`. In setup A I get a breakpoint trigger each tick, counting up in `latestChange`.  In setup B I get 3 triggers, one at the beginning, and two more at the end of the last craft. Notable is the value of 0 in `latestChange` in setup B.

**Comment by Mithi83** [comment]
This has been debugged and it turns out to be an AdvancedAE bug, not an AE2 bug. This ticket should be closed imho.
